### PR TITLE
New org permissions setup start page

### PR DESCRIPTION
**Pull request by benswannack**
## Context
We are updating the organisation permissions set up flow to be more user friendly

## Changes proposed in this pull request
Add a new component and view to display sorted and grouped organisation permissions before the flow begins
Set up the controller so that the flow can easily be powered by the same permissions list (this keeps the order consistent)

## Guidance to review
Test in the review app

Have a play with the component preview - although it doesn't do much: the sorting and grouping is done in the controller

## Link to Trello card
https://trello.com/c/5xdYki9b/3907-add-new-permissions-setup-start-page

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/controllers/provider_interface/organisation_permissions_setup_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module ProviderInterface
+  class OrganisationPermissionsSetupController < ProviderInterfaceController
+    before_action :require_manage_organisations_permission!
+    before_action :redirect_unless_permissions_to_setup
```
Comment (benswannack) [inline app/controllers/provider_interface/organisation_permissions_setup_controller.rb:4]: We probably also want to redirect to the old flow if the feature flag is off, and vice versa in the old controller

File: app/controllers/provider_interface/provider_interface_controller.rb [hunk 2]
```diff
@@ -97,17 +97,26 @@ def redirect_if_setup_required
...
-        redirect_to provider_interface_provider_relationship_permissions_organisations_path
+        redirect_to setup_organisation_permissions_path
       end
     end
 
     def performing_provider_organisation_setup?
       [
         ProviderInterface::ProviderAgreementsController,
         ProviderInterface::ProviderRelationshipPermissionsSetupController,
+        ProviderInterface::OrganisationPermissionsSetupController,
```
Comment (benswannack) [inline app/controllers/provider_interface/provider_interface_controller.rb:108]: If we use the feature flag here to decide which controller to use, then we don't need to redirect away in the individual controllers. Maybe that's better

File: config/routes.rb [hunk 3]
```diff
@@ -770,6 +770,14 @@
 
     resources :organisation_settings, path: '/organisation-settings', only: :index
 
+    resources :organisation_permissions_setup, only: %i[index edit update], path: 'organisation-permissions/setup' do
```
Comment (benswannack) [inline config/routes.rb:773]: Maybe we could switch this round and use a `setup` scope, and nest the resource in there, so the URL would be /setup/organisation-permissions

**Review by SamJamCul — APPROVED** [review]
Looks good, I'm interested to look at how we do the routes for the next set of pages, would be worth discussing

File: config/routes.rb [hunk 4]
```diff
@@ -770,6 +770,16 @@
 
     resources :organisation_settings, path: '/organisation-settings', only: :index
 
+    scope path: 'setup' do
+      resources :organisation_permissions_setup, only: %i[index edit update], path: 'organisation-permissions' do
```
Comment (benswannack) [inline config/routes.rb:774]: At the moment, we are only using the `index` route I've set up here.

I've added all the future routes here too so we can be consistent. In the flow, the URLs for each permission would be `/setup/organisation-permissions/org_permission_id/edit`. Then the results of that form are POSTed to `/setup/organisation-permissions/org_permission_id`. At the end, we GET `/setup/organisation-permissions/check`, POST to `/setup/organisation-permissions/commit` and then finally GET the success page at `/setup/organisation-permissions/success`

Comment (SamJamCul) [reply config/routes.rb:774]: Makes sense to me

**Pull request merged**